lms/djangoapps/certificates/management/commands/cert_generation.py

```diff
@@ -96,4 +97,7 @@ def handle(self, *args, **options):
                         user=user.id,
                         course=course_key
                     ))
-                generate_certificate_task(user, course_key)
+                try:
+                    generate_certificate_task(user, course_key)
+                except CertificateGenerationNotAllowed as e:
+                    raise CommandError(str(e)) from e
```

Comment:
In this case, instead of re-raising I would suggest we log it with WARN or ERROR level and move on to the next users. The command allows for a long list of users and the other scenarios where it fails (eg not enough grade) move on after logging so we should respect that.